src/Lean/Message.lean

```diff
@@ -78,12 +78,38 @@ inductive MessageData where
     Example: an inspector that tries to find "definitional equality failures" may look for the tag "DefEqFailure". -/
   | tagged            : Name → MessageData → MessageData
   | trace (data : TraceData) (msg : MessageData) (children : Array MessageData)
-  deriving Inhabited
+  /-- Lazy message data production. The `Dynamic` is expected to be `MessageData`, which
+  is a workaround for the positivity restriction. -/
```

Comment:
```suggestion
  /-- A lazy message.
  The provided thunk will not be run until it is about to be displayed.
  This can save computation in cases where the message may never be seen,
  e.g. when nested inside a collapsed trace.
  The `Dynamic` value is expected to be a `MessageData`,
  which is a workaround for the positivity restriction. -/
```